optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/solution/cloner/DeepCloningUtils.java

```diff
@@ -181,4 +187,56 @@ public boolean isFieldADeepCloneProperty(Field field, Class<?> owningClass) {
         }
         return false;
     }
+
+    /**
+     * @return never null
+     */
+    public Set<Class<?>> getDeepClonedTypeArguments(Type genericType) {
+        // Check the generic type arguments of the field.
+        // Yes, it is possible for fields and methods, but not instances!
```

Comment:
Let's leave prose and exclamation marks out of this, shall we? :-)

```suggestion
        // It is possible for fields and methods, but not instances.
```